pkg/build/strategies/sti/sti.go

```diff
@@ -620,11 +646,295 @@ func (builder *STI) Execute(command string, user string, config *api.Config) err
+
+	outReader, outWriter := io.Pipe()
+	defer outReader.Close()
+	defer outWriter.Close()
+
+	errReader, errWriter := io.Pipe()
+	defer errReader.Close()
+	defer errWriter.Close()
+
+	lastFileDstPath := "/tmp/upload-done"
+
+	waitForLastFileCmd := fmt.Sprintf("while [ ! -f %q ]; do sleep 0.5; done; rm -f %q", lastFileDstPath, lastFileDstPath)
+
+	opts := dockerpkg.RunContainerOptions{
+		Image:           builder.config.RuntimeImage,
+		PullImage:       false, // The PullImage is false because we've already pulled the image
+		CommandExplicit: []string{"/bin/sh", "-c", waitForLastFileCmd},
+		Stdout:          outWriter,
+		Stderr:          errWriter,
+		NetworkMode:     string(builder.config.DockerNetworkMode),
+		CGroupLimits:    builder.config.CGroupLimits,
+		CapDrop:         builder.config.DropCapabilities,
+		PostExec:        builder.postExecutor,
+		Env:             createBuildEnvironment(builder.config),
+		User:            "root", // without this we won't be able to remove last file
+	}
+
+	fd2, err := ioutil.TempFile("", "s2i-upload-done")
+	if err != nil {
+		return err
+	}
+	lastFilePath := fd2.Name()
+	defer func() {
+		fd2.Close()
+		os.Remove(lastFilePath)
+	}()
+
+	opts.OnStart = func(containerID string) error {
+		setStandardPerms := func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			// Skip chmod if on windows OS and for symlinks
+			if runtime.GOOS == "windows" || info.Mode()&os.ModeSymlink != 0 {
+				return nil
+			}
+			if info.IsDir() {
+				return os.Chmod(path, 0755)
+			}
+			// file should be writable by owner (u=w) and readable by other users (a=r),
+			// executable bit should be left as is
+			if info.Mode()&syscall.S_IEXEC != 0 {
+				return os.Chmod(path, 0755)
+			}
+			return os.Chmod(path, 0644)
+		}
+
+		workDir, err := builder.docker.GetImageWorkdir(builder.config.RuntimeImage)
+		if err != nil {
+			return fmt.Errorf("Couldn't get working dir of %q image: %v", builder.config.RuntimeImage, err)
+		}
+
+		artifactsDir := filepath.Join(builder.config.WorkingDir, api.RuntimeArtifactsDir)
+		glog.V(2).Infof("Uploading directory %q -> %q", artifactsDir, workDir)
+		if err := builder.docker.UploadToContainerWithCallback(artifactsDir, workDir, containerID, setStandardPerms, true); err != nil {
+			return fmt.Errorf("Couldn't upload directory (%q -> %q) into container %s: %v", artifactsDir, workDir, containerID, err)
+		}
+
+		glog.V(2).Infof("Uploading file %q -> %q", lastFilePath, lastFileDstPath)
+		if err := builder.docker.UploadToContainerWithCallback(lastFilePath, lastFileDstPath, containerID, setStandardPerms, true); err != nil {
+			return fmt.Errorf("Couldn't upload file (%q -> %q) into container %s: %v", lastFilePath, lastFileDstPath, containerID, err)
+		}
+
+		return err
+	}
+
+	go dockerpkg.StreamContainerIO(outReader, nil, func(message ...interface{}) {
+		glog.V(0).Infof("[runtime image stdout] %s", message...)
```

Comment:
so it turns out we are now defaulting the loglevel to 0, not 2, so anything we want printed all the time does need to be v(0).  sorry.

(whether this is actually something we want to print all the time, though, is another question...)